modules/trustpidSystem.js

```diff
@@ -0,0 +1,222 @@
+window.addEventListener('message', messageHandler, false);
+
+/**
+ * Get the "umid" from html5 local storage to make it available to the UserId module.
+ * @param config
+ * @returns {{trustpid: (*|string), acr: (string)}}
+ */
+function getTrustpidFromStorage() {
+  // Get the domain either from localStorage or global
+  let domain = JSON.parse(window.localStorage.getItem('fcIdConnectDomain')) || mnoDomain;
+  debugModeConsoleLog('info', `getTrustpid domain: ${domain}`);
+
+  if (!domain) {
+    debugModeConsoleLog('info', `getTrustpid domain: no domain`);
+    return {
+      trustpid: null,
+      acr: null,
+    };
+  }
+
+  // Get the acronym from global
+  let acronym = mnoAcronym;
+  // if acronym is empty, but "domain" is available, get the acronym from domain
+  if (!acronym) {
+    getDomainAcronym(domain);
+    acronym = mnoAcronym;
+  }
+
+  // Domain is correct in both local storage and idGraph, but no acronym is existing for the domain
+  if (domain && !acronym) {
+    return {
+      trustpid: null,
+      acr: null
+    }
+  }
+
+  let fcIdConnectObject;
+  let fcIdConnectData = JSON.parse(window.localStorage.getItem('fcIdConnectData'));
+  debugModeConsoleLog('info', `getTrustpid fcIdConnectData:
+  ${JSON.stringify(fcIdConnectData)}`);
+
+  if (fcIdConnectData &&
+    fcIdConnectData.connectId &&
+    Array.isArray(fcIdConnectData.connectId.idGraph) &&
+    fcIdConnectData.connectId.idGraph.length > 0) {
+    fcIdConnectObject = fcIdConnectData.connectId.idGraph.find(item => {
+      return item.domain === domain;
+    });
+  }
+  debugModeConsoleLog('info', `getTrustpid fcIdConnectObject:
+  ${JSON.stringify(fcIdConnectObject)}`);
+
+  return {
+    trustpid: (fcIdConnectObject && fcIdConnectObject.umid)
+      ? fcIdConnectObject.umid
+      : null,
+    acr: acronym,
+  };
+}
+
+/**
+ * Returns whether we are in Prebid debug mode or not.
+ * @returns {boolean}
+ */
+function isDebugMode() {
+  return (window.location.href.indexOf('?pbjs_debug=true') > -1) || (window.location.href.indexOf('&pbjs_debug=true') > -1);
+}
+
+/**
+ * Show debugging messages in browser developer console.
+ * @param type
+ * @param data
+ */
+function debugModeConsoleLog(type = 'info', data) {
+  if (isDebugMode()) {
+    switch (type) {
+      case 'info':
+        // eslint-disable-next-line no-console
+        console.info(data);
+        break;
+      case 'error':
+        // eslint-disable-next-line no-console
+        console.error(data);
+        break;
+      default:
+        // eslint-disable-next-line no-console
+        console.log(data);
+    }
+  }
+}
+
+/** @type {Submodule} */
+export const trustpidSubmodule = {
+  /**
+   * Used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+  /**
+   * Decodes the stored id value for passing to bid requests.
+   * @function
+   * @returns {{trustpid: string} | null}
+   */
+  decode(bidId) {
+    debugModeConsoleLog('info', `decode ${JSON.stringify(bidId)}`);
```

Comment:
instead of this method, I would just use `logInfo()` here

Reply:
Done